ASTNode.hpp

```diff
@@ -141,7 +141,60 @@ class ASTNode {
     // node will have an operator (e.g. +, *, etc.) specified somewhere (maybe
     // in the "literal"?) and one or two children run the child or children,
     // apply the operator to the returned value(s), then return the result
-    return 0;
+    if (literal == "!" || literal == "-"){
+      assert(children.size() == 1);
+      double left = children[0].RunExpect(symbols);
+      if (literal == "!"){
+        return left == 0 ? 1 : 0;
+      } else {
+        return -1 * left;
+      }
```

Comment:
There's no operators that should have zero children, so it's a bit cleaner if we pull out the repeated logic first (and also remove the `left` declaration for binary ops)

```suggestion
    assert(children.size() == 1);
    double left = children[0].RunExpect(symbols);
    if (literal == "!"){
      return left == 0 ? 1 : 0;
    }
    if (literal == "-"){
      return -1 * left;
    }
```